packages/fetchai/skills/tac_control_contract/helpers.py

```diff
@@ -0,0 +1,275 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the helpers methods for the controller agent."""
+
+import math
+import random
+from typing import Dict, List, Tuple, cast
+
+import numpy as np
+
+from aea.contracts.ethereum import Contract
+
+QUANTITY_SHIFT = 1  # Any non-negative integer is fine.
+TOKEN_TYPE = 2
+GAME_CURRENCY_TYPE = 2
+GAME_CURRENCY_ID = 2
+
+
+def generate_good_id_to_name(nb_goods: int, contract: Contract) -> Dict[str, str]:
```

Comment:
This is now implemented a mapping of good names to good ids, should be the other way round I believe?

Reply:
Note, you should get the ids from the contract and here just generate the mapping from id to name

Reply:
The generated ids that we are creating here based on the changes we have on the contract side. Returns me a list of integers. I don't no if I have to change something here based on the changes we made.